### PR TITLE
[chore] move oidcauth extension to generated lifecycle tests

### PR DESCRIPTION
**Pull request by atoulme**
Relates to https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/27849

Moved logic away from the constructor and into the config validate function.

### CODE REVIEW

File: extension/oidcauthextension/metadata.yaml [hunk 1]
```diff
@@ -7,3 +7,7 @@ status:
   distributions: [contrib, observiq, sumo]
   codeowners:
     active: [jpkrohling]
+
+tests:
+  config:
+  skip_lifecycle: true
```
Comment (dmitryax) [inline extension/oidcauthextension/metadata.yaml:13]: misaligned

File: extension/oidcauthextension/config.go [hunk 2]
```diff
@@ -30,3 +30,13 @@ type Config struct {
...
+
+func (c *Config) Validate() error {
+	if c.Audience == "" {
+		return errNoAudienceProvided
+	}
+	if c.IssuerURL == "" {
+		return errNoIssuerURL
+	}
+	return nil
+}
```
Comment (dmitryax) [inline extension/oidcauthextension/config.go:42]: Can be a separate PR?

File: extension/oidcauthextension/metadata.yaml (hunk 1 above)
Comment (atoulme) [reply extension/oidcauthextension/metadata.yaml:13]: `skip_lifecycle` is a flag outside of config to change the way we generate the test.

File: extension/oidcauthextension/config.go (hunk 2 above)
Comment (atoulme) [reply extension/oidcauthextension/config.go:42]: it could - in which case we should do it before this one.

Comment (atoulme) [reply extension/oidcauthextension/config.go:42]: https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/30460 is filed

File: extension/oidcauthextension/metadata.yaml (hunk 1 above)
Comment (dmitryax) [reply extension/oidcauthextension/metadata.yaml:13]: Oh, right. My bad

**Comment by dmitryax** [comment]
Please rebase

**Comment by atoulme** [comment]
Rebased.

**Review by dmitryax — APPROVED** [review]
